velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -258,6 +271,11 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
   }
 
   auto name = schemaElement.name;
+  if ((!options_.useColumnNamesForColumnMapping()) &&
+      (options_.fileSchema() != nullptr)) {
+    name = getNewFileSchemaColumnName(curSchemaIdx, name);
```

Comment:
DWRF is recursively rename field names in nested structs as well, I think parquet also probably want to do that